editor/ui2oh.cpp

```diff
@@ -324,8 +324,25 @@ bool MakeTimeTableSet(osmoh::OpeningHours const & oh, ui::TimeTableSet & tts)
   if (oh.IsTwentyFourHours())
     return true;
 
-  bool first = true;
+  // As "Closed" rules are not accepted if they are located at the beginning
```

Comment:
Why closed rules are not accepted?

Is it possible to convert "Mo off" text to the whole day closed rule?

Reply:
I haven't got into all the details of the opening hours (OH) parsing code, but here are my toughs/findings of what the problem might be:

The piece of code that rejects "Close" rules if they are at the start of the OH sequences is this one:

https://github.com/organicmaps/organicmaps/blob/7cf0b2c2a6b9146214b455c1b7c72f2b2c0e2d13/editor/ui2oh.cpp#L341-L345

I _presume_ that the OH parsing code starts with an empty OH sequence, then expects to have one or several opening hours rules and finally a list of "off hours" rules to subtract them from the previous opening hours rules That's why an "off" rule in the first place is useless and then the parser returns false.

For example, the sequence `Mo off; 9:00-17:00` any human would read it as: "On Mondays it's closed, and (for the rest of the days) it's open from 9 to 5".

But the OH parser does not like making these "assumptions", and expects to have first the actual opening hours, and later the "off" hours to subtract them: "9:00-17:00; Mo off" (open all days from 9 to 5, except for Mondays").

One way to avoid this problem is to move all the "off" rules at the end of the sequence, just what I've proposed in this PR.

Reply:
Thanks for the details!

[OSM spec](https://wiki.openstreetmap.org/wiki/Key:opening_hours) says in "Syntax for optional modifiers" section:
>     closed or off: closed at the indicated dates and times (or permanently if no time is given), except at overriding dates and times that follow;

So using the off modifier in the beginning is perfectly fine.

So maybe the author of the following code/comment was wrong, and removing that check (and maybe fixing other code if necessary) will provide a better fix?

```
    if (rulePart.GetModifier() == osmoh::RuleSequence::Modifier::Closed)
    {
      // off modifier in the first part in oh is useless.
      if (first == true)
        return false;
```